File: Examples/VisualBasic/ModifyingAndConvertingImages/DjVu/ConvertRangeOfDjVuPagesToSeparateImages.vb
Imports Aspose.Imaging.FileFormats.Djvu
Imports Aspose.Imaging.ImageOptions

'
'This project uses Automatic Package Restore feature of NuGet to resolve Aspose.Imaging for .NET API reference 
'when the project is build. Please check https:// Docs.nuget.org/consume/nuget-faq for more information. 
'If you do not wish to use NuGet, you can manually download Aspose.Imaging for .NET API from http://www.aspose.com/downloads, 
'install it and then add its reference to this project. For any issues, questions or suggestions 
'please feel free to contact us using http://www.aspose.com/community/forums/default.aspx
'

Namespace Aspose.Imaging.Examples.VisualBasic.ModifyingAndConvertingImages.DjVu
    Public Class ConvertRangeOfDjVuPagesToSeparateImages
        Public Shared Sub Run()
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_DjVu()

            ' Load a DjVu image
            Using image__1 As DjvuImage = DirectCast(Image.Load(dataDir & Convert.ToString("Sample.djvu")), DjvuImage)
                ' Create an instance of BmpOptions
                Dim exportOptions As New BmpOptions()
                ' Set BitsPerPixel for resultant images
                exportOptions.BitsPerPixel = 32
                ' Create an instance of IntRange and initialize it with range of pages to be exported
                Dim range As New IntRange(0, 2)
                ' Export first 2 pages
                Dim counter As Integer = 0
                For Each i In range.Range
                    ' Save each page in separate file, as BMP do not support layering
                    exportOptions.MultiPageOptions = New DjvuMultiPageOptions(range.GetArrayOneItemFromIndex(counter))
                    image__1.Save(dataDir & String.Format("{0}_out.bmp", System.Math.Max(System.Threading.Interlocked.Increment(counter), counter - 1)), exportOptions)
                Next
            End Using
        End Sub
    End Class
End Namespace

File: Examples/CSharp/ModifyingAndConvertingImages/DjVu/ConvertRangeOfDjVuPagesToSeparateImages.cs
using Aspose.Imaging.FileFormats.Djvu;
using Aspose.Imaging.ImageOptions;
using System;

/*
This project uses the Automatic Package Restore feature of NuGet to resolve the Aspose.Imaging for .NET API reference when the project is built. Please check https://learn.microsoft.com/en-us/nuget/resources/nuget-faq for more information.
If you do not wish to use NuGet, you can manually download Aspose.Imaging for .NET API from https://releases.aspose.com/, install it, and then add its reference to this project. For any issues, questions, or suggestions, please feel free to contact us via https://forum.aspose.com/
*/

namespace Aspose.Imaging.Examples.CSharp.ModifyingAndConvertingImages.DjVu
{
    public class ConvertRangeOfDjVuPagesToSeparateImages
    {
        public static void Run()
        {
            Console.WriteLine("Running example ConvertRangeOfDjVuPagesToSeparateImages");
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_DjVu();

            // Load a DjVu image
            using (DjvuImage image = (DjvuImage)Image.Load(dataDir + "Sample.djvu"))
            {
                // Create an instance of BmpOptions and set BitsPerPixel for the resultant images
                BmpOptions exportOptions = new BmpOptions();
                exportOptions.BitsPerPixel = 32;

                // Create an instance of IntRange and initialize it with the range of pages to be exported
                IntRange range = new IntRange(0, 2);
                int counter = 0;
                foreach (var i in range.Range)
                {
                    // Save each page in a separate file, as BMP does not support layering
                    exportOptions.MultiPageOptions = new DjvuMultiPageOptions(range.GetArrayOneItemFromIndex(counter));
                    image.Save(dataDir + string.Format("{0}_out.bmp", counter++), exportOptions);
                }
            }

            Console.WriteLine("Finished example ConvertRangeOfDjVuPagesToSeparateImages");
        }
    }
}